lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -389,4 +501,353 @@ private static ParquetCorruptionException corruptionException(String column, Par
+                else if (statistic.getMin() instanceof Float) {
+                    Integer min = floatToRawIntBits((Float) statistic.getMin());
+                    Integer max = floatToRawIntBits((Float) statistic.getMax());
+                    return canDropCanWithRangeStats(new ParquetIntegerStatistics((long) min, (long) max));
+                }
+                else if (statistic.getMin() instanceof Double) {
+                    Double min = (Double) statistic.getMin();
+                    Double max = (Double) statistic.getMax();
+                    return canDropCanWithRangeStats(new ParquetDoubleStatistics(min, max));
+                }
+                else if (statistic.getMin() instanceof Binary && columnDomain.getType() instanceof VarcharType) {
+                    Binary min = (Binary) statistic.getMin();
+                    Binary max = (Binary) statistic.getMax();
+                    return canDropCanWithRangeStats(new ParquetStringStatistics((Slices.wrappedBuffer(min.getBytes())), Slices.wrappedBuffer(max.getBytes())));
+                }
+                //TODO: Add other types
+            }
+            return false;
+        }
+
+        @Override
+        public boolean inverseCanDrop(org.apache.parquet.filter2.predicate.Statistics<T> statistics)
+        {
+            // !canDrop() cannot be used because it might not be correct. To be safe, we just keep the record by returning false.
+            // Since we don't use LogicalNotUserDefined, this method is not called.
+            return false;
+        }
+
+        private boolean canDropCanWithRangeStats(ParquetRangeStatistics parquetStatistics)
+        {
+            //TODO: hasNullValue is set true. Would it generate false negative?
+            Domain domain = createDomain(columnDomain.getType(), true, parquetStatistics);
+            if (columnDomain.intersect(domain).isNone()) {
+                return true;
+            }
+            return false;
+        }
+    }
+
+    class ColumnIndexValueConverter
+    {
+        private final Map<String, Function<Object, Object>> ciConversions;
+
+        private ColumnIndexValueConverter(List<RichColumnDescriptor> columns)
+        {
+            this.ciConversions = new HashMap<>();
+            for (RichColumnDescriptor column : columns) {
+                ciConversions.put(column.getPrimitiveType().getName(), getColumnIndexConversions(column.getPrimitiveType()));
+            }
+        }
+
+        public List<Long> getMinValuesAsLong(Type type, ColumnIndex columnIndex, String column)
+        {
+            int pageCount = columnIndex.getMinValues().size();
+            List<ByteBuffer> minValues = columnIndex.getMinValues();
+            List<Long> mins = new ArrayList<>();
+            for (int i = 0; i < pageCount; i++) {
+                if (TINYINT.equals(type) || SMALLINT.equals(type) || INTEGER.equals(type)) {
+                    int minValue = converter.convert(minValues.get(i), column);
```

Comment:
I'd probably inline the various `minValue`s and `maxValue`s in this file that are immediately added to a list and never used again.